cucumber/api/features/authenticate.feature

```diff
@@ -20,6 +20,19 @@ Feature: Exchange a role's API key for a signed authentication token
       cucumber:user:alice successfully authenticated with authenticator authn
     """
 
+    Scenario: X-Request-Id is set and visible in the audit record
+    Then I set the "X-Request-Id" header to "TestMyApp"
```

Comment:
out of curiosity - what's the use-case here? how will this benefit a real user?

Reply:
I believe it was just an undocumented/test `behavior`. It could be helpful during debugging, it would allow us to easily find the logs that are correlated with a request. CC: @alexkalish